apps/web/app/layoutHOC.tsx

```diff
@@ -30,17 +34,31 @@ export function WithLayout<T extends Record<string, any>>({
       props = (await getData(buildLegacyCtx(h, cookies(), p.params, p.searchParams))) ?? ({} as T);
     }
 
-    const children = "children" in p ? p.children : null;
+    // `p.children` exists only for layout.tsx files
+    const childrenFromLayoutFile = "children" in p ? p.children : null;
+    const page = ServerPage ? (
+      await ServerPage({ ...props, ...p })
+    ) : Page ? (
+      <Page {...props} />
```

Comment:
FYI, there's no need to pass `p` object (which has `params` and `searchParams` inside) to `Page` because this Page is a client component and it uses hooks like useSearchParams for such props.